src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/AvaloniaXamlIlLanguage.cs

```diff
@@ -185,6 +185,15 @@ public IXamlCustomAttribute GetCustomAttribute(IXamlProperty property, IXamlType
         public static bool CustomValueConverter(AstTransformationContext context,
             IXamlAstValueNode node, IXamlType type, out IXamlAstValueNode result)
         {
+            if (node is AvaloniaXamlIlOptionMarkupExtensionTransformer.OptionsMarkupExtensionNode optionsNode)
+            {
+                if (optionsNode.ConvertToReturnType(context, type, out var newOptionsNode))
+                {
+                    result = newOptionsNode;
+                    return true;
+                }
+            }
```

Comment:
Normally compiler would emit type convertion code and try to parse/convert in runtime.
This custom converter will do it compile time.

Alternatively I was thinking to introduce XamlX . IXamlConvertableValueNode with `ConvertToType` method and same signature. But not sure how useful it would be.

Reply:
We could keep that as is for now and generalize it later if we need it elsewhere.